{
  ProjectWorkspace workspace=TestDataHelper.createProjectWorkspaceForScenario(this,"class_statements",temporaryFolder);
  workspace.setUp();
  ProjectWorkspace.ProcessResult result=workspace.runBuckCommand("test","--all");
  result.assertTestFailure("Tests should fail");
  String stderr=result.getStderr();
  assertThat("Explanatory exception output is present",stderr,containsString("BOOM!"));
  assertThat("Some tests still pass",stderr,containsRegex(createBuckTestOutputLineRegex("PASS",3,0,0,"com.example.EverythingOKTest")));
  assertThat(stderr,containsRegex(createBuckTestOutputLineRegex("FAIL",0,0,1,"com.example.HasBeforeClassFailure")));
  Pattern failPattern=Pattern.compile("^FAIL +(\\d+)ms",Pattern.MULTILINE);
  Matcher matcher=failPattern.matcher(stderr);
  matcher.find();
  int actualRuntime=Integer.parseInt(matcher.group(1));
  assertThat("Reasonable runtime was guessed",actualRuntime,greaterThanOrEqualTo(BEFORE_CLASS_EXPECTED_RUNTIME));
}
